modules/safer-cluster/main.tf

```diff
@@ -0,0 +1,166 @@
+
+  maintenance_start_time = var.maintenance_start_time
+
+  initial_node_count = var.initial_node_count
+
+  // We suggest removing the default node pull, as it cannot be modified without
+  // destroying the cluster.
+  remove_default_node_pool = true
+
+  disable_legacy_metadata_endpoints = true
+
+  node_pools        = var.node_pools
+  node_pools_labels = var.node_pools_labels
+
+  // TODO(mmontan): check whether we need to restrict these
+  // settings.
+  node_pools_metadata = var.node_pools_metadata
+  node_pools_taints   = var.node_pools_taints
+  node_pools_tags     = var.node_pools_tags
+
+  // TODO(mmontan): we generally considered applying
+  // just the cloud-platofrm scope and use Cloud IAM
+  // If we have Workload Identity, are there advantages
+  // in restricting scopes even more?
+  node_pools_oauth_scopes = var.node_pools_oauth_scopes
+
+  stub_domains         = var.stub_domains
+  upstream_nameservers = var.upstream_nameservers
+
+  // We should use IP Alias.
+  configure_ip_masq = false
+
+  logging_service    = var.logging_service
+  monitoring_service = var.monitoring_service
+
+  // We never use the default service account for the cluster. The default
+  // project/editor permissions can create problems if nodes were to be ever
+  // compromised.
+
+  // We either:
+  // - Create a dedicated service account with minimal permissions to run nodes.
+  //   All applications shuold run with an identity defined via Workload Identity anyway.
+  // - Use a service account passed as a parameter to the module, in case the user
+  //   wants to maintain control of their service accounts.
+  create_service_account = length(var.compute_engine_service_account) > 0 ? false : true
+  service_account        = var.compute_engine_service_account
+
+  // TODO(mmontan): define a registry_project parameter in the private_beta_cluster,
+  // so that we can give GCS permissions to the service account on a project
+  // that hosts only container-images and not data.
+  grant_registry_access = true
+
+  // Basic Auth disabled
+  basic_auth_username = ""
+  basic_auth_password = ""
+
+  issue_client_certificate = false
+
+  cluster_ipv4_cidr = var.cluster_ipv4_cidr
+
+  cluster_resource_labels = var.cluster_resource_labels
+
+  // We enable private endpoints to limit exposure.
+  enable_private_endpoint       = true
+  deploy_using_private_endpoint = true
+
+  // Private nodes better control public exposure, and reduce
+  // the ability of nodes to reach to the Internet without
+  // additional configurations.
+  enable_private_nodes = true
+
+  master_ipv4_cidr_block = var.master_ipv4_cidr_block
+
+  // Istio is recommended for pod-to-pod communications.
+  istio    = var.istio
+  cloudrun = var.cloudrun
+
+  default_max_pods_per_node = var.default_max_pods_per_node
+
+  database_encryption = var.database_encryption
+
+  // We suggest to define policies about  which images can run on a cluster.
+  enable_binary_authorization = true
```

Comment:
It's worth noting that since you are using the `beta-private-cluster`, this field `enable_binary_authorization` is actually not implemented yet, it's just a stub so it doesn't do anything